### PR TITLE
Feature/flex 4410 lmd improvements batch

### PR DESCRIPTION
**Pull request by rlemmers**

### CODE REVIEW

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/AdHocManagementService.java [hunk 1]
```diff
@@ -125,14 +126,14 @@ public void getStatus(final String organisationIdentification, final String devi
...
-                ? DeviceFunction.GET_LIGHT_SENSOR_STATUS.name()
-                : messageType;
+                ? DeviceFunction.GET_LIGHT_SENSOR_STATUS.name() : messageType;
 
         this.osgpCoreRequestMessageSender.send(new RequestMessage(correlationUid, organisationIdentification,
                 deviceIdentification, allowedDomainTypeDto), actualMessageType, messagePriority, device.getIpAddress());
     }
 
-    public void handleGetStatusResponse(final org.opensmartgridplatform.dto.valueobjects.DeviceStatusDto deviceStatusDto,
+    public void handleGetStatusResponse(
```
Comment (jenkins-ip-10-4-24-184) [inline osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/AdHocManagementService.java]: MAJOR Method has 9 parameters, which is greater than 6 authorized. [rule](https://10.4.24.184/sonarqube/coding_rules#rule_key=squid%3AS00107)

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    @Test
+    public void rejectDifferentBatchNumbers() {
+        final CdmaDevice cd1 = new CdmaDevice("cd1", this.loopbackAddress, "200/1", (short) 1);
+        final CdmaDevice cd2 = new CdmaDevice("cd2", this.loopbackAddress, "200/1", (short) 2);
+
+        final CdmaBatch batch = new CdmaBatch(cd1);
+        try {
+            batch.addCdmaDevice(cd2);
+            Assert.fail("Different batch numbers should be rejected");
+        } catch (final IllegalArgumentException e) {
```
Comment (smvdheijden) [inline osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java]: Consider using `@Test(expected = IllegalArgumentException.class)` in stead of empty catch

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Test
+    public void newCdmaBatch() {
+        final CdmaDevice cd1 = new CdmaDevice("cd1", this.loopbackAddress, "200/1", (short) 1);
+        final CdmaDevice cd2 = new CdmaDevice("cd2", this.loopbackAddress, "200/1", (short) 1);
+
+        final CdmaBatch batch = new CdmaBatch(cd1);
+        batch.addCdmaDevice(cd2);
+
+        Assert.assertEquals(batch.getBatchNumber(), cd1.getBatchNumber());
```
Comment (smvdheijden) [inline osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java]: Consider using static imports for Assert.assertEquals, Assert.fails etc. (Applies to all test classes)

No message? ;-)

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Test
+    public void equalsWhenBatchNumbersMatch() {
+        final CdmaDevice cdBatch1 = new CdmaDevice("cd1", this.loopbackAddress, "200/1", (short) 1);
+        final CdmaDevice cdBatch2 = new CdmaDevice("cd2", this.loopbackAddress, "333/5", (short) 1);
+
+        final CdmaBatch batch1 = new CdmaBatch(cdBatch1);
+        final CdmaBatch batch2 = new CdmaBatch(cdBatch2);
+        Assert.assertEquals("Batches with the same batch number should be equal", batch1, batch2);
```
Comment (smvdheijden) [inline osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java]: Is this really wanted behavior? To me it makes more sense for a batch to be equal if the mast segment is equal as well.

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Test
+    public void largerWhenBatchNumberLarger() {
+        final CdmaDevice cdBatch1 = new CdmaDevice("cd1", this.loopbackAddress, "200/1", (short) 3);
+        final CdmaDevice cdBatch2 = new CdmaDevice("cd2", this.loopbackAddress, "200/1", (short) 2);
+
+        final CdmaBatch batch1 = new CdmaBatch(cdBatch1);
+        final CdmaBatch batch2 = new CdmaBatch(cdBatch2);
+        Assert.assertNotEquals("Batches with different batch numbers should not be equal", batch1, batch2);
+        Assert.assertTrue(batch1.compareTo(batch2) > 0);
```
Comment (smvdheijden) [inline osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java]: No message? ;-)

It would be nice to include tests with null values as well

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaMastSegmentTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Test
+    public void popCdmaBatch() {
+        final CdmaDevice cd1 = new CdmaDevice("cd1", this.loopbackAddress, "200/1", (short) 1);
+        final CdmaDevice cd2 = new CdmaDevice("cd2", this.loopbackAddress, "200/1", (short) 1);
+        final CdmaMastSegment mastSegment = new CdmaMastSegment(cd1);
+        mastSegment.addCdmaDevice(cd2);
+
+        final CdmaBatch batch = mastSegment.popCdmaBatch();
+        Assert.assertEquals(batch.getCdmaBatchDevices().size(), 2);
```
Comment (smvdheijden) [inline osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaMastSegmentTest.java]: No message? ;-) (Applies to assertTrue below as well)

Swap arguments (first expected, then actual) to be compliant with sonar rule:
Assertion arguments should be passed in the correct order (squid:S3415)

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaMastSegmentTest.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+public class CdmaMastSegmentTest {
+
+    private final InetAddress loopbackAddress = InetAddress.getLoopbackAddress();
+
+    @Test
+    public void newCdmaMastSegment() {
+        final CdmaDevice cd1 = new CdmaDevice("cd1", this.loopbackAddress, "200/1", (short) 1);
+        final CdmaMastSegment mastSegment = new CdmaMastSegment(cd1);
+
+        Assert.assertEquals(mastSegment.getMastSegment(), "200/1");
```
Comment (smvdheijden) [inline osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaMastSegmentTest.java]: No message? ;-) (Applies to assertTrue below as well)

Swap arguments (first expected, then actual) to be compliant with sonar rule:
Assertion arguments should be passed in the correct order (squid:S3415)

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaRunTest.java [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+    @Test
+    public void itemPerMastSegment() {
+        final CdmaRun run = new CdmaRun();
+        for (int i = 0; i < 5; i++) {
+            run.add(new CdmaDevice("cd" + i, this.loopbackAddress, "200/" + i, (short) 1));
+        }
+
+        final Iterator<CdmaMastSegment> iterator = run.getMastSegmentIterator();
+        try {
```
Comment (smvdheijden) [inline osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaRunTest.java]: Is this try catch needed here?

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/AdHocManagementService.java [hunk 9]
```diff
@@ -125,14 +126,14 @@ public void getStatus(final String organisationIdentification, final String devi
...
+            final org.opensmartgridplatform.dto.valueobjects.DeviceStatusDto deviceStatusDto,
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/AdHocManagementService.java]: Done.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/AdHocManagementService.java [hunk 10]
```diff
@@ -320,14 +289,27 @@ public void handleLightMeasurementDeviceTransition(final String organisationIden
...
-        final List<Ssld> ssldsToTransition = this.getSsldsToTransitionForLmd(lmd);
+        // Retrieve the devices per mast/segment
+        final CdmaRun cdmaRun = this.getCdmaDevicesPerMastSegment(lmd);
 
         // Determine the transition type based on the event of the LMD.
         final TransitionType transitionType = this.determineTransitionTypeForEvent(event);
 
         // Send SET_TRANSITION messages to the SSLDs.
-        this.transitionSslds(ssldsToTransition, organisationIdentification, correlationUid, transitionType);
+        this.setTransitionService.transitionCdmaRun(cdmaRun, organisationIdentification, correlationUid,
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/AdHocManagementService.java]: Renamed to setTransitionForCdmaRun.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaDeviceList.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2018 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.domain.publiclighting.application.services.transition;
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaDeviceList.java]: Deleted the class, it wasn't used.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaDeviceList.java [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(this.iteration, this.mastSegment, this.batchNumber);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (!(obj instanceof CdmaDeviceList)) {
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaDeviceList.java]: Comment no longer relevant, because I deleted the class.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java [hunk 13]
```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright 2018 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.domain.publiclighting.application.services.transition;
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java]: SetTransitionService was the only file in the new "transition" package. I've moved the class one level up, to org.opensmartgridplatform.adapter.domain.publiclighting.application.services and deleted the "transition" package.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+            try {
+                if (inetAddress != null) {
+                    SetTransitionService.this.transitionDevice(metadata, inetAddress.getHostAddress(), transitionType,
+                            emptyTransitionTime);
+                } else {
+                    LOGGER.warn("setTransition not possible, because InetAddress is null for device {}",
+                            metadata.getDeviceIdentification());
+                }
+            } catch (final Exception e) {
+                LOGGER.warn("Set transition failed failed for " + metadata, e);
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java]: Done.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/valueobjects/CdmaBatch.java [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+            return true;
+        }
+
+        if (!(obj instanceof CdmaBatch)) {
+            return false;
+        }
+
+        final CdmaBatch other = (CdmaBatch) obj;
+
+        return Objects.equals(this.batchNumber, other.batchNumber);
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/valueobjects/CdmaBatch.java]: I refactored the code, so CdmaBatch no longer uses the CdmaDevice class. In the new situation, a CdmaBatch doesn't have any information about segments whatsoever. The equal on just batch number now seems appropriate.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/valueobjects/CdmaBatchDevice.java [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.domain.publiclighting.application.valueobjects;
+
+import java.net.InetAddress;
+
+import org.opensmartgridplatform.domain.core.valueobjects.CdmaDevice;
+
+public class CdmaBatchDevice {
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/valueobjects/CdmaBatchDevice.java]: It contains only two fields instead of four. Segment and batch are not relevant for the device itself.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/valueobjects/CdmaRun.java [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+        }
+    }
+
+    public Iterator<CdmaMastSegment> getMastSegmentIterator() {
+        return this.mastSegments.values().iterator();
+    }
+
+    @Override
+    public String toString() {
+        final List<String> segmentNames = this.mastSegments.values().stream().map(CdmaMastSegment::getMastSegment)
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/valueobjects/CdmaRun.java]: Done.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/valueobjects/CdmaMastSegment.java [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        final CdmaMastSegment other = (CdmaMastSegment) obj;
+
+        return Objects.equals(this.mastSegmentName, other.mastSegmentName);
+    }
+
+    @Override
+    public String toString() {
+        final List<String> batchNumbers = this.cdmaBatches.values().stream().map(x -> x.getBatchNumber().toString())
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/valueobjects/CdmaMastSegment.java]: Done (and simplified further).

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+                LOGGER.info("Send messages for {}", cdmaBatch);
+                final List<CdmaBatchDevice> devices = cdmaBatch.getCdmaBatchDevices();
+
+                // Send a message for each device
+                devices.forEach(device -> MastSegmentRunnable.this.transitionDeviceRunnable(device,
+                        organisationIdentification, correlationUid, transitionType));
+            });
+        }
+
+        private void transitionDeviceRunnable(final CdmaBatchDevice device, final String organisationIdentification,
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java]: Renamed to setTransitionForDeviceRunnable.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+
+    private final Integer delayBetweenBatchSeconds;
+    private final ScheduledExecutorService executor;
+
+    public SetTransitionService(final ScheduledExecutorService executor, final int delayBetweenBatchSeconds) {
+        this.executor = executor;
+        this.delayBetweenBatchSeconds = delayBetweenBatchSeconds;
+    }
+
+    public void transitionDevice(final MessageMetadata metadata, @NotNull final TransitionType transitionType,
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java]: Renamed to setTransitionForDevice.

File: osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java [hunk 21]
```diff
@@ -0,0 +1,188 @@
...
+        LOGGER.debug("Public setTransitionForDevice called for device {} with organisation {}", deviceIdentification,
+                organisationIdentification);
+
+        this.findOrganisation(organisationIdentification);
+        final Device device = this.findActiveDevice(deviceIdentification);
+
+        this.transitionDevice(metadata, device.getIpAddress(), transitionType, transitionTime);
+    }
+
+    private void transitionDevice(final MessageMetadata metadata, final String ipAddress,
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/SetTransitionService.java]: Renamed to setTransitionForDevice.

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java (hunk 2 above)
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaBatchTest.java]: I deleted the test. It's no longer relevant after my refactoring of CdmaBatch.

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaMastSegmentTest.java (hunk 6 above)
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaMastSegmentTest.java]: Done.

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaMastSegmentTest.java (hunk 7 above)
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaMastSegmentTest.java]: Done.

File: osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaRunTest.java (hunk 8 above)
Comment (rlemmers) [reply osgp-adapter-domain-publiclighting/src/test/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/services/transition/CdmaRunTest.java]: It's not needed. If there's an error during the loop, the test will fail as well. I've removed the try/catch.

File: osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.domain.core.valueobjects;
+
+import java.net.InetAddress;
+import java.util.Objects;
+
+public class CdmaDevice {
+
+    // private static final Short MAX_BATCH_NUMBER = (short) 99;
```
[the remark this answers is not in the document]
Comment (smvdheijden) [reply osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java]: Remove commented code

File: osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+    // public static final String DEFAULT_MASTSEGMENT =
+    // "DEVICE-WITHOUT-MASTSEGMENT";
```
[the remark this answers is not in the document]
Comment (smvdheijden) [reply osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java]: Remove commented code

File: osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java [hunk 24]
```diff
@@ -0,0 +1,76 @@
...
+    private InetAddress networkAddress;
+    private String mastSegmentName;
+    private Short batchNumber;
+
+    public CdmaDevice(final String deviceIdentification, final InetAddress networkAddress, final String mastSegmentName,
+            final Short batchNumber) {
+        this.deviceIdentification = deviceIdentification;
+        this.networkAddress = networkAddress;
+        this.mastSegmentName = mastSegmentName; // == null ? DEFAULT_MASTSEGMENT
+                                                // : mastSegmentName;
```
[the remark this answers is not in the document]
Comment (smvdheijden) [reply osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java]: Clean up code

File: osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java [hunk 25]
```diff
@@ -0,0 +1,76 @@
...
+        this.batchNumber = batchNumber; // == null ? MAX_BATCH_NUMBER :
+                                        // batchNumber;
```
[the remark this answers is not in the document]
Comment (smvdheijden) [reply osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java]: Clean up code

File: osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java (hunk 23 above)
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java]: Done.

File: osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java (hunk 24 above)
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java]: Done.

File: osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java (hunk 22 above)
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java]: Done.

File: osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java (hunk 25 above)
[the remark this answers is not in the document]
Comment (rlemmers) [reply osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/CdmaDevice.java]: Done.
